tests/components/shelly/test_config_flow.py

```diff
@@ -101,72 +90,34 @@ async def test_form(hass, gen):
     assert result2["title"] == "Test name"
     assert result2["data"] == {
         "host": "1.1.1.1",
-        "model": "SHSW-1",
+        "model": model,
         "sleep_period": 0,
         "gen": gen,
     }
     assert len(mock_setup.mock_calls) == 1
     assert len(mock_setup_entry.mock_calls) == 1
 
 
-async def test_title_without_name(hass):
-    """Test we set the title to the hostname when the device doesn't have a name."""
-
-    result = await hass.config_entries.flow.async_init(
-        DOMAIN, context={"source": config_entries.SOURCE_USER}
-    )
-    assert result["type"] == data_entry_flow.FlowResultType.FORM
-    assert result["errors"] == {}
-
-    settings = MOCK_SETTINGS.copy()
-    settings["name"] = None
-    settings["device"] = settings["device"].copy()
-    settings["device"]["hostname"] = "shelly1pm-12345"
-    with patch(
-        "homeassistant.components.shelly.config_flow.get_info",
-        return_value={"mac": "test-mac", "type": "SHSW-1", "auth": False},
-    ), patch(
-        "aioshelly.block_device.BlockDevice.create",
-        new=AsyncMock(
-            return_value=Mock(
-                model="SHSW-1",
-                settings=settings,
-            )
-        ),
-    ), patch(
-        "homeassistant.components.shelly.async_setup", return_value=True
-    ) as mock_setup, patch(
-        "homeassistant.components.shelly.async_setup_entry",
-        return_value=True,
-    ) as mock_setup_entry:
-        result2 = await hass.config_entries.flow.async_configure(
-            result["flow_id"],
-            {"host": "1.1.1.1"},
-        )
-        await hass.async_block_till_done()
-
-    assert result2["type"] == data_entry_flow.FlowResultType.CREATE_ENTRY
-    assert result2["title"] == "shelly1pm-12345"
-    assert result2["data"] == {
-        "host": "1.1.1.1",
-        "model": "SHSW-1",
-        "sleep_period": 0,
-        "gen": 1,
-    }
-    assert len(mock_setup.mock_calls) == 1
-    assert len(mock_setup_entry.mock_calls) == 1
-
-
 @pytest.mark.parametrize(
     "test_data",
     [
-        (1, {"username": "test user", "password": "test1 password"}, "test user"),
-        (2, {"password": "test2 password"}, "admin"),
+        (
+            1,
```

Comment:
Side note: I'd recommend putting each item as its own parameter. Then you can add type annotations for each parameter, making the the test more clear.

Reply:
https://github.com/home-assistant/core/pull/86778